Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIMenusCocoa.mm

```diff
@@ -353,6 +381,69 @@
     return *m_onClicked;
 }
 
+void WebExtensionContextProxy::dispatchMenusClickedEvent(const WebExtensionMenuItemParameters& menuItemParameters, bool wasChecked, const WebExtensionMenuItemContextParameters& contextParameters, const std::optional<WebExtensionTabParameters>& tabParameters)
+{
+    // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/menus/onClicked
+
+    ASSERT(menuItemParameters.type);
+
+    auto *info = [NSMutableDictionary dictionary];
+
+    info[menuIDKey] = toMenuIdentifierWebAPI(menuItemParameters.identifier);
+
+    if (menuItemParameters.parentIdentifier)
+        info[parentMenuIDKey] = toMenuIdentifierWebAPI(menuItemParameters.parentIdentifier.value());
+
+    if (isCheckedType(menuItemParameters.type.value())) {
+        info[checkedKey] = @(menuItemParameters.checked.value_or(false));
+        info[wasCheckedKey] = @(wasChecked);
+    }
+
+    if (!contextParameters.selectionString.isNull())
+        info[selectionTextKey] = contextParameters.selectionString;
+
+    if (!contextParameters.sourceURL.isNull())
+        info[srcURLKey] = contextParameters.sourceURL.string();
+
+    if (!contextParameters.linkURL.isNull())
+        info[linkURLKey] = contextParameters.linkURL.string();
```

Comment:
If we are passing a string - could this be something like `linkURLStringKey`?

Reply:
This is for the result being passed to JS, and string the only option.